app/helpers/tabs_helper.rb

```diff
@@ -1,107 +1,88 @@
 # encoding: utf-8
+# html used in tabsets
 module TabsHelper
-
   # Short-hand to render shared tab_set partial for a given set of links.
   def draw_tab_set(links)
-    render(partial: "/shared/tab_set", locals: {links: links})
+    render(partial: "/shared/tab_set", locals: { links: links })
   end
 
-  # Draw the cutesy eye icons in the upper right side of screen.  It does it
-  # by creating a "right" tab set.  Thus this must be called in the header of
-  # the view and must not actually be rendered.  Typical usage would be:
-  #
-  #   # At top of view:
-  #   <%
-  #     # Specify the page's title.
-  #     @title = "Page Title"
-  #
-  #     # Define set of linked text tabs for top-left.
-  #     new_tab_set do
-  #       add_tab("Tab Label One", link: args, ...)
-  #       add_tab("Tab Label Two", link: args, ...)
-  #       ...
-  #     end
-  #
-  #     # Draw interest icons in the top-right.
-  #     draw_interest_icons(@object)
-  #   %>
-  #
-  # This will cause the set of three icons to be rendered floating in the
-  # top-right corner of the content portion of the page.
-  #
-  def draw_interest_icons(object)
-    if @user
-      type = object.type_tag
+  # assemble HTML for "tabset" for show_observation
+  # actually a list of links and the interest icons
+  def show_observation_tabset(obs, user)
+    tabs = []
 
-      # Create link to change interest state.
-      def interest_link(label, object, state) # :nodoc:
-        link_with_query(label,
-          controller: :interest,
-          action: :set_interest,
-          id: object.id,
-          type: object.class.name,
-          state: state
-        )
-      end
+    tabs << google_images_and_dist_map_links(obs)
+    tabs << general_questions_link(obs, user)
+    tabs << notifications_link(obs, user)
+    tabs << manage_lists_link(obs, user)
+    tabs << map_link
+    tabs << obs_change_links(obs)
 
-      # Create large icon image.
-      def interest_icon_big(type, alt) # :nodoc:
-        image_tag("#{type}2.png",
-          alt: alt,
-          width: "50px",
-          height: "50px",
-          class: "interest_big",
-          title: alt
-        )
-      end
+    tabs << draw_interest_icons(obs)
 
-      # Create small icon image.
-      def interest_icon_small(type, alt) # :nodoc:
-        image_tag("#{type}3.png",
-          alt: alt,
-          width: "23px",
-          height: "23px",
-          class: "interest_small",
-          title: alt
-        )
-      end
+    { pager_for: obs, right: draw_tab_set(tabs.reject(&:empty?)) }
+  end
 
-      def interest_tab(img1, img2, img3)
-        content_tag(:div, img1 + safe_br + img2 + img3, class: "interest-eyes")
-        # content_tag(:div, img1 + safe_br + img2 + img3, style: "position: absolute; top: 0;")
-      end
+  def google_images_and_dist_map_links(obs)
+    return unless obs.try(:name).known?
```

Comment:
Jason is right.  It is better to keep the models as simple as possible (just providing data accessors and such) and keep the HTML stuff in another location.  Another case of old bad habits emerging.  It does seem like we have enough Google stuff that we could separate it out into it's own thing.  A better approach might be to just consider Google one instance of third party URL providers and have a module for that.  Given that there are clearly two distinct examples, that's enough these days for me to think it should go in its own module especially since I know we use a Google services in other places and we have the links to MycoPortal etc.  So you might end up with something name_image_link(:google, name) and name_map_link(:google, name).  The :google stuff could even be a backend configuration option so you'd end up with just name_image_link(name), name_map_link(name) and leave it up to those functions to decide who the provider is going to be.  Sort of depends on how far you want to go.